pkg/lib/authenticationflow/session_bot_protection.go

```diff
@@ -0,0 +1,13 @@
+package authenticationflow
+
+type BotProtectionVerificationStatus string
```

Comment:
We should not store a string only. For example, when we verify successfully with the provider, we want to store the verification response as well. So it should be a struct. Since it is a struct, and sometimes we do not need it, the context value should be a pointer to the struct. We need not to define BotProtectionVerificationStatusDefault anymore. Since a nil pointer to the struct means initial status.

```golang
type BotProtectionVerificationResult struct {
  Outcome BotProtectionVerificationOutcome `json:"outcome,omitempty"`
  Response interface{} `json:"response,omitempty"`
}

type BotProtectionVerificationOutcome string

const (
  BotProtectionVerificationOutcomeFailed = "failed"
  BotProtectionVerificationOutcomeVerified = "verified"
  BotProtectionVerificationOutcomeServiceUnavailable = "service_unavailable"
)
```

Reply:
Or you omit this change, and add it until you actual implement the verification.

Reply:
> Or you omit this change, and add it until you actual implement the verification.

Prefer include in this PR 🙏 rationale - It's not "non-implemented feature`, it's referenced in some intent/node reactions in next PR

> `interface{}`

How about `any`? ref https://github.com/golang/go/issues/49884

Reply:
Just keep using the good old `interface{}`